compiler/rustc_parse/src/lexer/tokentrees.rs

```diff
@@ -246,16 +246,22 @@ impl<'a> TokenTreesReader<'a> {
 
     #[inline]
     fn parse_token_tree_other(&mut self) -> TokenTree {
+        // `spacing` for the returned token is determined by the next token:
+        // its kind and its `preceded_by_whitespace` status.
```

Comment:
Maybe expand this comment a bit to explain that "spacing is ws *after* the token", the semantics is not obvious just from the lexer's layer itself.